src/main/java/com/dabsquared/gitlabjenkins/GitLabPushTrigger.java

```diff
@@ -101,11 +102,11 @@
     private transient boolean acceptMergeRequestOnSuccess;
     private transient MergeRequestLabelFilter mergeRequestLabelFilter;
 
-
+    @Deprecated
```

Comment:
Did something happen to the empty constructor you created? I was only mentioning to use the `@deprecated` annotation only if there was a new constructor which should be used instead of this one, and this one should be deprecated in favor of the new constructor + using setter methods. If the other empty constructor is not needed, then you should remove the deprecated annotation. I also believe that adding a `@deprecated` should require javadoc suggesting to use the other constructor.

Reply:
Ah, apologies. I stripped the merge back to only what's required to support setters and left the rest for fear of affecting backwards compatibility.

I will add no-arg constructors where necessary and update `@Deprecated` constructors to inform clients to use the no-arg version in conjunction with setters.